src/plugin/nerd_plugin.js

```
@@ -97,37 +105,80 @@ class NerdPluginVirtual extends EventEmitter {
       settleIfOver: this.settleIfOver,
       initialBalance: opts.initialBalance
     })
+
     this.balance.on('_balanceChanged', (balance) => {
       this._log('balance changed to ' + balance)
       this.emit('_balanceChanged', balance)
-      this._sendBalance()
+      this.rpc.notify({ type: 'balance', balance: balance })
     })
+
     this.balance.on('under', (balance) => {
-      this._log('you should settle your balance of ' + balance)
-      this._sendSettle()
-    })
-    this.balance.on('over', (balance) => {
-      this._log('you should settle your balance of ' + balance)
-      this.emit('settlement', balance)
+      return co.wrap(this._settle).call(this, balance)
     })
 
-    if (this.settler) {
-      this.on('settlement', (balance) => {
-        if (!this.settleAddress) return
+    this.balance.on('over', (balance) => {
+      this._log('settling your balance of ' + balance)
+      if (!this.settleAddress || !this.settler) return
```

I'd log something whenever you fail silently.